test/ConsensusApp/validTransition.test.ts

```diff
@@ -3,134 +3,107 @@ import {ethers} from 'ethers';
-      noValidTransitionError,
-    );
-  });
-
-  it('valid veto', async () => {
-    const variablePartOld = constructConsensusVariablePart(2, defaultOutcome1, defaultOutcome2);
-    const variablePartNew = constructConsensusVariablePart(0, defaultOutcome1, emptyOutcome);
-    expect(
-      await consensusApp.validTransition(variablePartOld, variablePartNew, defaultTurn, 3),
-    ).toBe(true);
-  });
+const description00 = 'valid consensus -> propose';
+const description01 = 'invalid consensus -> propose: furtherVotesRequired too low';
+const description02 = 'valid vote';
+const description03 = 'invalid vote: furtherVotesRequired not decreased';
+const description04 = 'valid veto';
+const description05 = 'invalid veto: proposedOutcome not empty';
+const description06 = 'valid pass';
+const description07 = 'invalid pass: currentOutcome1 ≠ currentOutcome2';
+const description08 = 'valid finalVote';
+const description09 = 'invalid finalVote: proposedOutcome1 ≠ currentOutcome2';
+const description10 = 'invalid finalVote: proposedOutcome not empty';
 
-  it('invalid veto: currentOutcome1 ≠ currentOutcome2', async () => {
-    const variablePartOld = constructConsensusVariablePart(2, defaultOutcome1, defaultOutcome2);
-    const variablePartNew = constructConsensusVariablePart(0, defaultOutcome2, emptyOutcome);
-    await expectRevert(
-      () => consensusApp.validTransition(variablePartOld, variablePartNew, defaultTurn, 3),
-      noValidTransitionError,
+describe('validTransition', () => {
+  it.each`
+    description      | furtherVotesRequired | outcome           | proposedOutcome   | reason
+    ${description00} | ${[0, 2]}            | ${['0x1', '0x1']} | ${['0x', '0x1']}  | ${undefined}
+    ${description01} | ${[0, 1]}            | ${['0x1', '0x1']} | ${['0x', '0x1']}  | ${noValidTransitionError}
+    ${description02} | ${[2, 1]}            | ${['0x1', '0x1']} | ${['0x1', '0x1']} | ${undefined}
+    ${description03} | ${[2, 2]}            | ${['0x1', '0x1']} | ${['0x1', '0x1']} | ${noValidTransitionError}
+    ${description04} | ${[2, 0]}            | ${['0x1', '0x1']} | ${['0x2', '0x']}  | ${undefined}
+    ${description05} | ${[2, 0]}            | ${['0x1', '0x2']} | ${['0x2', '0x']}  | ${noValidTransitionError}
+    ${description06} | ${[0, 0]}            | ${['0x1', '0x1']} | ${['0x', '0x']}   | ${undefined}
+    ${description07} | ${[0, 0]}            | ${['0x1', '0x2']} | ${['0x', '0x']}   | ${noValidTransitionError}
+    ${description08} | ${[1, 0]}            | ${['0x1', '0x2']} | ${['0x2', '0x']}  | ${undefined}
+    ${description09} | ${[1, 0]}            | ${['0x1', '0x3']} | ${['0x2', '0x']}  | ${noValidTransitionError}
+    ${description10} | ${[1, 0]}            | ${['0x1', '0x2']} | ${['0x2', '0x2']} | ${noValidTransitionError}
+  `('$description', async ({furtherVotesRequired, outcome, proposedOutcome, reason}) => {
+    const variablePartOld = constructConsensusVariablePart(
```

Comment:
Nit, we should probably remain consistent between the contract and the testsuite and either use `a` to `b` nomenclature or `old` to `new` nomenclature.

Reply:
Agreed. I went for a/b for now (since that is in use elsewhere in the contracts). fb218e3